Kernel/Locking/Mutex.cpp

```diff
@@ -249,19 +251,19 @@ ErrorOr<void> Mutex::block(Thread& current_thread, Mode mode, SpinlockLocker<Spi
     return {};
 }
 
-void Mutex::unblock_waiters(Mode previous_mode)
+ErrorOr<void> Mutex::unblock_waiters(Mode previous_mode)
```

Comment:
I understand it was a `FIXME` that drove this change, but in this case the `FIXME` pointed you in the wrong direction. I don't think this is the correct route to go down. Mutex locking needs to be designed in a way that can't fail. Having a `Mutex::unlock()` that can fail makes it impossible to reason about the state of the system.

Reply:
It was my own `FIXME` from a previous commit lol. Should I just delete this commit then and revert to `MUST`ing those unblocks?

Reply:
If you aren't interested in in fixing the allocations in unblock, then yea. Might be nice to add a FIXME to those functions so it's clear that we need to fix their implementations to not allocate.